infrastructure/variables.tf

```diff
@@ -13,3 +13,11 @@ variable "region" {
 variable "zone" {
   default = "europe-west1-b"
 }
+
+variable "secrets" {
+  default = {
+    "optic-firebase-project-id"   = "FIREBASE_PROJECT_ID"
+    "optic-firebase-client-email" = "FIREBASE_CLIENT_EMAIL"
+    "optic-firebase-private-key"  = "FIREBASE_PRIVATE_KEY_BASE64"
+  }
+}
```

Comment:
Q: if we are using tf to provision the secrets, do we really need to create them in the cd workflow? if we don't, then the code in the workflow is largely simplified as we only have to update the value, instead of deleting and creating each secret

Reply:
I couldn't find a workflow that was supposed to run terraform. That's why I didn't create the secrets in the terraform code, but rather assumed they would be created by the time somebody runs terraform (if we tried to create a secret in terraform that is already there, terraform would fail - so people would need to import the secrets manually when running terraform). But sure we can do the other way around.

My point (and this is in answer to both comments) is that we didn't want to keep adding new secret versions since they cost money and they can't be updated. What Piotr suggested is that we could retrieve the secrets on the workflow, check their value, and if there are no changes we would just let the workflow continue. If there are changes, we create a new version of the secret. After that, we could destroy the old one.

So, to summarize:

- Create the secrets in terraform
- Only create new secret versions if the values have changed
- Delete old versions

Does it all make sense?